core/src/test/scala/unit/kafka/server/AbstractCreateTopicsRequestTest.scala

```diff
@@ -152,15 +151,6 @@ class AbstractCreateTopicsRequestTest extends BaseRequestTest {
   protected def error(error: Errors, errorMessage: Option[String] = None): ApiError =
     new ApiError(error, errorMessage.orNull)
 
-  protected def toStructWithDuplicateFirstTopic(request: CreateTopicsRequest): Struct = {
```

Comment:
Interesting that this is no longer used. Do you know if we are still testing the case where we have a duplicate topic?

Reply:
I tracked the removal of the uses of this method back to this: https://github.com/apache/kafka/commit/e2e8bdbd8cb6ca2ac962c72147d21a9e8b9ba2c0#diff-217dea05d3ff55c8bf10503c329f7c2b. I don't see a clear reason for it.

Reply:
@cmccabe do you remember why the test was removed?

Reply:
CreateTopicsRequest was the first RPC converted over to automatic generation, if I remember correctly.  I converted it while writing the generator.

I think when the automatic code generation was first developed, it didn't have support for multiple identical elements in the same map.  So it wasn't possible to serialize the invalid request at that point.  So I had to remove this test.  Later on I added support for duplicate entries in the same map.

We should definitely re-add a test for duplicate createTopics elements.  There should also be a test like this in KafkaAdminClientTest to test the admin client's handling of it (AC should not allow this to even get to the server)

Reply:
Can we file a JIRA please?